### ISSUE TITLE
Configure failure: missing include symlink for Python 2.7

### ISSUE THREAD

**Issue by MincingWords** [apple/turicreate#553]
#### Description
Configure failure: missing include symlink for Python 2.7

#### Steps/Code to Reproduce
unzip master.zip
cd turicreate-master
./configure

#### Expected Results
Configure completes normally

#### Actual Results
When running ./configure in preparation to build, it fails saying 

> /opt/turicreate-master
> + linux_patch_sigfpe_handler
> + [[ linux-gnu == linux* ]]
> + targfile=deps/local/include/pyfpe.h
> + echo '#undef WANT_SIGFPE_HANDLER'
> + cat - deps/local/include/pyfpe.h
> cat: deps/local/include/pyfpe.h: No such file or directory

And higher up we see this has happened:

> + mkdir -p deps/local/lib
> + mkdir -p deps/local/include
> + pushd deps/local/include
> /opt/turicreate-master/deps/local/include /opt/turicreate-master
> ++ ls '../../env/include/python2.7/*'
> ls: cannot access '../../env/include/python2.7/*': No such file or directory

By that time the script has made .../deps/env/include and put a symlink in there to /usr/include/python3.6m called python3.6m but it needs to also create one to /usr/include/python2.7 called python2.7. Create that symlink by hand, go back up to the distribution root and do ./configure again and it will work. This is on Gentoo Linux and its order of preference for Python is currently 3.6, 3.4, 2.7, and 3.5.

#### Versions
>>> import platform; print(platform.platform())
Linux-4.9.76-gentoo-r1-x86_64-Intel-R-_Core-TM-_i5-4570_CPU_@_3.20GHz-with-gentoo-2.4.1
>>> import sys; print("Python", sys.version)
Python 3.6.5 (default, Apr 18 2018, 09:46:48) 
[GCC 6.4.0]
>>> import numpy; print("NumPy", numpy.__version__)
NumPy 1.10.4
>>> import scipy; print("SciPy", scipy.__version__)
SciPy 0.19.1
>>> import sklearn; print("Scikit-Learn", sklearn.__version__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'sklearn'

**Comment by znation** [comment]
@TobyRoseman I'm guessing this is caused by lack of virtualenv (I suspect we don't really check for its existence, we just rely on it and fail to error out early enough if it's missing). The fix might be to check for virtualenv earlier and error out with a useful message.

**Comment by znation** [comment]
Nevermind, this is what you get if you don't have virtualenv:
```
6692b6d7aa2a /turicreate # ./configure
Installing python toolchain.
python version = python2.7.
/turicreate /turicreate
+ set -e
+ [[ -z '' ]]
+ VIRTUALENV=virtualenv
+ PIP=pip3
+ PYTHON_FULL_NAME=python2.7m
+ [[ python2.7 == \p\y\t\h\o\n\2\.\7 ]]
+ PIP=pip
+ PYTHON_FULL_NAME=python2.7
+ virtualenv /turicreate/deps/env
./scripts/install_python_toolchain.sh: line 19: virtualenv: command not found
6692b6d7aa2a /turicreate # 
```

**Comment by znation** [comment]
Oh, I think this is what happens if you don't have libpython2.7-dev.

**Comment by znation** [comment]
Ok, I think the root cause of the issue here is that nothing is checking whether `--python[2.7|3.5|3.6]` supplied to `./configure` (or lack thereof, which is 2.7) matches the actual Python version. I think we should be using the [`--python=`](https://virtualenv.pypa.io/en/stable/reference/#cmdoption-p) option on virtualenv to make the environment's Python version match our `./configure` script.

**Comment by znation** [comment]
Looks like this will be fixed with the fix for #494.
